pkg/model/utxo/unspent.go

```
@@ -35,21 +63,49 @@ func (u *Manager) IsOutputUnspent(outputID *iotago.UTXOInputID) (bool, error) {
 	return u.IsOutputUnspentWithoutLocking(output)
 }
 
-func (u *Manager) ForEachUnspentOutputWithoutLocking(consumer OutputConsumer, address ...*iotago.Ed25519Address) error {
+func (u *Manager) ForEachUnspentOutput(consumer OutputConsumer, options ...UTXOIterateOption) error {
+
+	opt := iterateOptions(options)
+
+	if opt.readLockLedger {
+		u.ReadLockLedger()
+		defer u.ReadUnlockLedger()
+	}
 
 	var innerErr error
 
 	key := []byte{UTXOStoreKeyPrefixUnspent}
-	if len(address) > 0 {
-		if len(address[0]) != iotago.Ed25519AddressBytesLength {
-			return ErrInvalidAddressSize
+
+	// Filter by address
+	if opt.address != nil {
+		addrBytes, err := opt.address.Serialize(iotago.DeSeriModeNoValidation)
+		if err != nil {
+			return err
 		}
-		key = byteutils.ConcatBytes(key, address[0][:])
+		key = byteutils.ConcatBytes(key, addrBytes)
+	}
+
+	// Filter by type
+	if opt.filterOutputType != nil {
```

Same problem like in the spents file.
Output type filter can only be applied if address filter was set.
Maybe return an error if someone tries to filter by output type without address filter set?